### PR TITLE
[Java] [JDBC] Flight JDBC Driver Tests Failed Binary Accessors 

### PR DESCRIPTION
**Pull request by iurysalino**
[Refactor Tests about Binary Accessors to return SQLException]

### CODE REVIEW

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java [hunk 1]
```diff
@@ -167,23 +157,8 @@ public double getDouble() {
...
-      return buffer.putShort((short) value).array();
-    } else if (bytesToAllocate == Integer.BYTES) {
-      return buffer.putInt((int) value).array();
-    } else if (bytesToAllocate == Long.BYTES) {
-      return buffer.putLong(value).array();
-    }
-
-    throw new UnsupportedOperationException();
+  public byte[] getBytes() throws SQLException {
+    throw new SQLException();
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java:161]: Missing message on exception

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 2]
```diff
@@ -62,35 +64,34 @@
         };
         if (vector instanceof UInt1Vector) {
           return new ArrowFlightJdbcBaseIntVectorAccessor((UInt1Vector) vector, getCurrentRow,
-              noOpWasNullConsumer);
+          noOpWasNullConsumer);
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:67]: Wrong formatting?

**Review by rafael-telles — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 2 above)
Comment (jcralmeida) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:67]: Also checking this format on the rest of the methods in this if statement

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 3]
```diff
@@ -125,124 +126,104 @@ public static void setup() {
...
-  @Test
+  @Test(expected = SQLException.class)
   public void testShouldGetBytesFromIntVector() throws Exception {
     byte[] value = new byte[] {(byte) 0xaa, (byte) 0xbb, (byte) 0xcc, (byte) 0xdd};
 
-    accessorIterator
-        .assertAccessorGetter(intVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+    accessorIterator.assertAccessorGetter(intVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
+        CoreMatchers.is(value));
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:144]: Change this to use the new assert method that throws an exception

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 4]
```diff
@@ -125,124 +126,104 @@ public static void setup() {
...
-  @Test
+  @Test(expected = SQLException.class)
   public void testShouldGetBytesFromSmallVector() throws Exception {
     byte[] value = new byte[] {(byte) 0xaa, (byte) 0xbb};
-
-    accessorIterator
-        .assertAccessorGetter(smallIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+    accessorIterator.assertAccessorGetter(smallIntVector,
+        ArrowFlightJdbcBaseIntVectorAccessor::getBytes, CoreMatchers.is(value));
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:211]: Change this to use the new assert method that throws an exception

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 5]
```diff
@@ -125,124 +126,104 @@ public static void setup() {
...
-  @Test
+  @Test(expected = SQLException.class)
   public void testShouldGetBytesFromTinyIntVector() throws Exception {
     byte[] value = new byte[] {(byte) 0xaa};
-
-    accessorIterator
-        .assertAccessorGetter(tinyIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+    accessorIterator.assertAccessorGetter(tinyIntVector,
+        ArrowFlightJdbcBaseIntVectorAccessor::getBytes, CoreMatchers.is(value));
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:218]: Change this to use the new assert method that throws an exception

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 6]
```diff
@@ -125,124 +126,104 @@ public static void setup() {
...
         new byte[] {(byte) 0xaa, (byte) 0xbb, (byte) 0xcc, (byte) 0xdd, (byte) 0xee, (byte) 0xff,
-            (byte) 0xaa,
-            (byte) 0xbb};
-
-    accessorIterator
-        .assertAccessorGetter(bigIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+            (byte) 0xaa, (byte) 0xbb};
+    accessorIterator.assertAccessorGetter(bigIntVector,
+        ArrowFlightJdbcBaseIntVectorAccessor::getBytes, CoreMatchers.is(value));
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:227]: Change this to use the new assert method that throws an exception

**Review by jcralmeida — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java (hunk 1 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java:161]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 2 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:67]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 6 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:227]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 5 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:218]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 4 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:211]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 3 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:144]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java (hunk 2 above)
Comment (jcralmeida) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java]: Not fixed yet

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 7]
```diff
@@ -125,124 +124,97 @@ public static void setup() {
...
 
   @Test
-  public void testShouldGetBytesFromIntVector() throws Exception {
-    byte[] value = new byte[] {(byte) 0xaa, (byte) 0xbb, (byte) 0xcc, (byte) 0xdd};
-
-    accessorIterator
-        .assertAccessorGetter(intVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+  public void testShouldGetBytesFromIntVectorThrowSqlException() throws Exception {
+    accessorIterator.assertAccessorGetterForException(intVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes);
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:139]: Change the name to `assertAccessorGetterThrowingException`

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 8]
```diff
@@ -125,124 +124,97 @@ public static void setup() {
...
   }
 
   @Test
-  public void testShouldGetBytesFromSmallVector() throws Exception {
-    byte[] value = new byte[] {(byte) 0xaa, (byte) 0xbb};
-
-    accessorIterator
-        .assertAccessorGetter(smallIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+  public void testShouldGetBytesFromSmallVectorThrowSqlException() throws Exception {
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:203]:
```suggestion
  public void testShouldGetBytesFromSmallVectorThrowsSqlException() throws Exception {
```

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 9]
```diff
@@ -125,124 +124,97 @@ public static void setup() {
...
-  public void testShouldGetBytesFromBigIntVector() throws Exception {
-    byte[] value =
-        new byte[] {(byte) 0xaa, (byte) 0xbb, (byte) 0xcc, (byte) 0xdd, (byte) 0xee, (byte) 0xff,
-            (byte) 0xaa,
-            (byte) 0xbb};
-
-    accessorIterator
-        .assertAccessorGetter(bigIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+  public void testShouldGetBytesFromBigIntVectorThrowSqlException() throws Exception {
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:215]:
```suggestion
  public void testShouldGetBytesFromBigIntVectorThrowsSqlException() throws Exception {
```

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBitVectorAccessorTest.java [hunk 10]
```diff
@@ -64,7 +63,8 @@ public void tearDown() {
     this.vectorWithNull.close();
   }
 
-  private <T> void iterate(final Function<ArrowFlightJdbcBitVectorAccessor, T> function,
+  private <T> void iterate(final AccessorTestUtils
+      .AccessorIterator.CheckedFunction<ArrowFlightJdbcBitVectorAccessor, T> function,
```
[the remark this answers is not in the document]
Comment (jcralmeida) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBitVectorAccessorTest.java:67]:
```suggestion
  private <T> void iterate(final CheckedFunction<ArrowFlightJdbcBitVectorAccessor, T> function,
```

**Review by jcralmeida — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java (hunk 1 above)
Comment (rafael-telles) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessor.java:161]: Not fixed?

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/text/ArrowFlightJdbcVarCharVectorAccessor.java [hunk 11]
```diff
@@ -163,8 +163,8 @@ public double getDouble() throws SQLException {
   public BigDecimal getBigDecimal() throws SQLException {
     try {
       return new BigDecimal(this.getString());
-    } catch (Exception e) {
-      throw new SQLException(e);
+    } catch (NumberFormatException exception) {
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/text/ArrowFlightJdbcVarCharVectorAccessor.java:166]: Not related to this PR?

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 12]
```diff
@@ -125,124 +124,97 @@ public static void setup() {
...
+  public void testShouldGetBytesFromIntVectorThrowsSqlException() throws Exception {
+    accessorIterator.assertAccessorGetterThrowingException(intVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes);
   }
 
   @Test
-  public void testShouldGetBytesFromIntVectorWithNull() throws Exception {
-    accessorIterator.assertAccessorGetter(intVectorWithNull,
-        ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-        CoreMatchers.nullValue());
+  public void testShouldGetBytesFromIntVectorWithNullThrowsSqlException() throws Exception {
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:143]: Redundant test, can be removed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java [hunk 13]
```diff
@@ -125,124 +124,97 @@ public static void setup() {
...
-    byte[] value =
-        new byte[] {(byte) 0xaa, (byte) 0xbb, (byte) 0xcc, (byte) 0xdd, (byte) 0xee, (byte) 0xff,
-            (byte) 0xaa,
-            (byte) 0xbb};
-
-    accessorIterator
-        .assertAccessorGetter(bigIntVector, ArrowFlightJdbcBaseIntVectorAccessor::getBytes,
-            CoreMatchers.is(value));
+  public void testShouldGetBytesFromBigIntVectorThrowsSqlException() throws Exception {
+
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBaseIntVectorAccessorUnitTest.java:216]: nit: Remove blank line

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBitVectorAccessorTest.java [hunk 14]
```diff
@@ -43,9 +45,9 @@
           (vector, getCurrentRow) -> new ArrowFlightJdbcBitVectorAccessor((BitVector) vector,
               getCurrentRow, (boolean wasNull) -> {
           });
-  private final AccessorTestUtils.AccessorIterator<ArrowFlightJdbcBitVectorAccessor>
+  private final AccessorIterator<ArrowFlightJdbcBitVectorAccessor>
       accessorIterator =
-      new AccessorTestUtils.AccessorIterator<>(collector, accessorSupplier);
+      new AccessorIterator<>(collector, accessorSupplier);
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcBitVectorAccessorTest.java:50]: Looks like it has checkstyle issues

**Review by rafael-telles — CHANGES_REQUESTED** [review]

**Review by rafael-telles — CHANGES_REQUESTED** [review]

**Pull request merged**